lib/tests/streamlit/runtime/state/widgets_test.py

```diff
@@ -423,10 +461,6 @@ def test_widget_id_computation_no_form_widgets(self, widget_func, module_name):
         sig = inspect.signature(widget_func)
         expected_sig = self.signature_to_expected_kwargs(sig)
 
-        # button and chat widgets don't include a form_id param in their calls to
-        # compute_and_register_element_id because having either in forms (aside from the form's
-        # submit button) is illegal.
-        del expected_sig["form_id"]
```

Comment:
Do we not need this anymore because `form_id` is not passed for buttons anymore in the first place?

Reply:
It's irrelevant for most buttons if `form_id` is used or not. Since most buttons cannot be used in form it would anyways always be empty... but if we enable some other buttons for forms (e.g. download button would be a good candidate) they would need the form ID. And the `form_submit_button` (variation of the button) already requires the `form_id`.

Reply:
Alrighty, makes sense 👍 although `download_button` in form probably does not make too much sense right? because if you click on it, it cannot take into account what is in the form right now because its not submitted. And then it also doesn't make sense to have in in the form at all

Reply:
The current download button directly starts the download in click, (theoretically) not requiring a rerun. So it could be used in a form similar to how `link_button` could be used. In practice, we still do a partly useless rerun which is something we are considering removing.

Reply:
Btw. I changed it back to only add form ID for form submit button. But that doesn't change something with this test since its still required to be passed in with `None`.